### ISSUE TITLE
Collection is getting deleted automatically from qdrant vector db in production environment.

### ISSUE THREAD

**Issue by yashna200023** [qdrant/qdrant#3445]
The created collection in the qdrant is getting deleted automatically. 

Currently , I have hosted the qdrant  instance in GCP and the collections are getting deleted automatically .But the collections in qdrant in local host environment its seems to be fine and doesnt face the issue.

**Qdrant server version : v1.7.3**

**GCP Resources:**
1.Memory 32GiB
2.CPU 8

This is the code I am using to create the collection:

```
def create_collection(collection_name):
    url = f'{qdrant_url}/collections/{collection_name}'
    headers = {
        'Authorization': f'Bearer {qdrant_api_key}',
        'Content-Type': 'application/json'
    }
    payload = {"vectors": {"size": 1536, "distance": "Cosine"}}
    response = requests.put(url, json=payload, headers=headers)

    if response.status_code == 200:
        print("Collection created successfully")
    else:
        print(f"Error: {response.status_code}, {response.text}")
```

**Comment by timvisee** [comment]
Your creation request looks fine. Qdrant never deletes a collection on its own.

Are you sure your instance has storage properly configured? In other words, are you sure that if your Docker container restarts all data is properly kept?

By the way, since you're using Python we recommend to use our Python client instead: https://github.com/qdrant/qdrant-client

**Comment by kalki7** [comment]
Even I've got the same issue but locally, the data does persistent on the local volume that's been mounted. A volume mount is all I'm using and I've noticed that a container restart, doesn't have any issues. It's only after a docker restart do have I noticed a data loss.

**Comment by timvisee** [comment]
It may be related to a similar problem. Do you have the same issue with other containers that store data? What kind of configuration are you running, is it Docker on Windows?

**Comment by kalki7** [comment]
Apologies for the earlier lapse in the service name. The data directory is just a local path on the desktop, which is the exact same as I used for other volume mounts in the past for scalar databases.

**Comment by timvisee** [comment]
> Apologies for the earlier lapse in the service name. The data directory is just a local path on the desktop, which is the exact same as I used for other volume mounts in the past for scalar databases.

Right! And if you view this directory yourself you do see data being created, which stays after the container/docker is brought down?

**Comment by kalki7** [comment]
> > Apologies for the earlier lapse in the service name. The data directory is just a local path on the desktop, which is the exact same as I used for other volume mounts in the past for scalar databases.
> 
> Right! And if you view this directory yourself you do see data being created, which stays after the container/docker is brought down?

Yes, I'm able to see the data getting created, updated. The interesting thing is that the data persists on my local volume, it's just that Qdrant wasn't able to identify it. But if i recreate the same index again, the old data on the volume mounts gets overwritten.

**Comment by timvisee** [comment]
It must have something to do with Qdrant not being able to read the files from disk. But I'm not sure how that can ahppen.

**Comment by kalki7** [comment]
Could it be a file permission thing ?

**Comment by timvisee** [comment]
> Could it be a file permission thing ?

Yes, but then I'd expect to see an error showing us this.

Are you seeing any warning or error messages in the Qdrant logs? Just before stopping and right when starting?

**Comment by kalki7** [comment]
Nothing of that sort, they look clean

**Comment by Mageswaran1989** [comment]
I too ran it to this today. The data was there before system restart. After system reboot, I see zero docs.

**Comment by lemuriandezapada** [comment]
Just had a similar issue. System restart lead to database being wiped

**Comment by davide445** [comment]
Happened to me either, but in a very bizarre sequence of events: suddenly all my older collections disappeared as soon I created a new one Friday. The new collections created this weekend was working fine.
This morning I was using the latest Sunday created collection and suddenly (no reboot, no nothing) this collection was gone, but I got back al the older collections that previously disappeared.

**Comment by timvisee** [comment]
> Happened to me either, but in a very bizarre sequence of events: suddenly all my older collections disappeared as soon I created a new one Friday. The new collections created this weekend was working fine. This morning I was using the latest Sunday created collection and suddenly (no reboot, no nothing) this collection was gone, but I got back al the older collections that previously disappeared after a restart.

It sounds like this can only happen if persisted storage is messed up. Maybe due to an incorrect mount, or due to mixing up persisted volumes with wrong nodes.

**Comment by davide445** [comment]
> > Happened to me either, but in a very bizarre sequence of events: suddenly all my older collections disappeared as soon I created a new one Friday. The new collections created this weekend was working fine. This morning I was using the latest Sunday created collection and suddenly (no reboot, no nothing) this collection was gone, but I got back al the older collections that previously disappeared after a restart.
> 
> It sounds like this can only happen if persisted storage is messed up. Maybe due to an incorrect mount, or due to mixing up persisted volumes with wrong nodes.

My problem is happening on my testing/prototyping Ubuntu PC, where there is only two containers running, one Qdrant when I start it and the second Portainer (upped months ago, just stopped). The only reason I can find is I was starting every time a different container when I started Qdrant, and I was stopping it using ctrl+C, so maybe there was some inconsistency problem building up in the persistent storage. Still, I find strange the behavior.
I have now named the Qdrant container so to start always the same and stop it using Docker. 
If the problem will pop out again I will shift to a regular Qdrant installation.

**Comment by davide445** [comment]
The problem showed up again, all old collections disappeared and only last created one remains, but empty.
I just followed Qdrant documentation to install and launch Qdrant docker. 
Seems installing a binary it's not a supported option (require building your own from the source code). As written I have nothing else working on docker, nor have I setup anything that can change the storage behavior (literally copied and pasted Qdrant doc commands).
I cannot waste time and have demos failing due Qdrant decide suddently to make collections disappear, so considering passing to anoth

**Comment by timvisee** [comment]
@davide445 Could you check whether your collections survive:
1. restarting the Qdrant container
2. restarting your script to upload/search in Qdrant

If point 1 fails, you didn't set up mounts correctly. If so, please feel free to share how you set the container up.
If point 2 fails, you might accidentally be deleting the collection on start. That can be due to a `recreate_collection` call.

Please note that our cloud is free of these problems. You can get a free instance here: https://cloud.qdrant.io/

**Comment by timvisee** [comment]
Assuming this is now resolved.
